### PR TITLE
chore: upgrades coraza docker

### PR DESCRIPTION
**Pull request by sts**
- Run coraza-spoa as unprivileged user within the container
- Device images in a main image without CRS and a coreruleset target to include CRS
- Add multiarch images for multiple platforms
- Use docker-entrypoint.sh
- Use tini

### CODE REVIEW

**Comment by jcchavezs** [comment]
Do you mind installing the pre-commit hook and running the linter?

**Comment by jcchavezs** [comment]
Can you please rebase?

**Comment by fzipi** [comment]
@jcchavezs rebased and fixed some stuff... can you take a look?

File: .github/workflows/container-image.yaml [hunk 1]
```diff
@@ -41,17 +41,13 @@ jobs:
...
-        uses: docker/metadata-action@v4
-        with:
-          images: |
-            ghcr.io/${{ github.repository }}
-          tags: |
-            type=ref,event=branch
-            type=ref,event=pr
-            type=semver,pattern={{version}}
-            type=semver,pattern={{major}}
+      - name: Inspect builder
```
Comment (jcchavezs) [inline .github/workflows/container-image.yaml:44]: Curious about this change, is it needed?

File: .github/workflows/container-image.yaml [hunk 2]
```diff
@@ -60,12 +56,56 @@ jobs:
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Build and push
+      - name: Docker metadata - Main
+        id: meta1
```
Comment (jcchavezs) [inline .github/workflows/container-image.yaml:60]: Can we use more descriptive names for `metaX`?

File: Dockerfile [hunk 3]
```diff
@@ -1,59 +1,81 @@
...
-    # Rename coraza-spoa default config file
-    && mv /app/config.yaml.default /app/config.yaml \
-    # Rename coraza-spoa binary
-    && mv /app/coraza-spoa_amd64 /app/coraza-spoa \
-    # Make coraza-spoa binary executable
-    && chmod +x /app/coraza-spoa
+# Switch to root for crs installation
+USER root
 
+# Download the core rule set
```
Comment (jcchavezs) [inline Dockerfile:67]: While this is outside the scope of the PR in the future we might want to use coraza-coreruleset.

File: docker/coraza-spoa/docker-entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+
```
Comment (jcchavezs) [inline docker/coraza-spoa/docker-entrypoint.sh:2]: Shall we add a comment on why this is needed?

File: Dockerfile [hunk 5]
```diff
@@ -1,59 +1,81 @@
...
+RUN --mount=type=cache,target=/root/.cache/go-build \
+    --mount=type=cache,target=/go/pkg \
+    OS=${TARGETOS} ARCH=${TARGETARCH} make
+
+# ---
+FROM alpine:3.17 AS main
+
+ARG TARGETARCH
+
+LABEL org.opencontainers.image.authors="OWASP Coraza Maintainers" \
```
Comment (jcchavezs) [inline Dockerfile:27]: Shall we use `The OWASP Coraza contributors` to align with other repos?

File: .github/workflows/container-image.yaml (hunk 1 above)
Comment (fzipi) [reply .github/workflows/container-image.yaml:44]: @sts ?

I think this is mostly informational.

File: docker/coraza-spoa/docker-entrypoint.sh (hunk 4 above)
Comment (sts) [reply docker/coraza-spoa/docker-entrypoint.sh:2]: 👍🏻

File: .github/workflows/container-image.yaml (hunk 1 above)
Comment (sts) [reply .github/workflows/container-image.yaml:44]: yes, this is just to include information about the build in the ci output.

File: .github/workflows/container-image.yaml (hunk 2 above)
Comment (sts) [reply .github/workflows/container-image.yaml]: Changed in latest commit

File: Dockerfile (hunk 5 above)
Comment (sts) [reply Dockerfile]: Changed in latest commit

**Pull request merged**